docs/kb/semgrep-ci/jenkins-shared-library-with-semgrep.md

```diff
@@ -0,0 +1,119 @@
+---
+tags:
+  - Jenkins
+  - Shared Library
+description: Setting up Jenkins shared library with Semgrep scans
+---
+
+# Jenkins shared library with Semgrep scans
+
+## Motivation
+A good practice when programming is to avoid duplicate code. The main issue with code duplication is not the duplication itself, but when you need to modify a piece of this duplicated code, you need to do it two, three or n-times as many repetitions as you have. And even worse, when the duplicated fragment has a bug or a vulnerability, you need to fix it in all the occurrences.
+Luckily, there is an easy solution for this: encapsulate code in only one place: a method, a function, or a library; it doesn’t matter; the key concept here is removing duplications and replacing them with a simple call.
```

Comment:
I'd probably keep this shorter and say something like "A good practice when programming is to avoid duplicate code. This applies even when defining your build pipelines. To prevent duplicate code when defining Semgrep tests with Jenkins, create a [Jenkins Shared Library](https://www.jenkins.io/doc/book/pipeline/shared-libraries/) with the Semgrep snippet. Then, call this library in all the projects you have in your organization.

Reply:
Ok, thanks!